collector/netdev_linux.go

```diff
@@ -50,12 +50,14 @@ func parseNetDevStats(r io.Reader, ignore *regexp.Regexp) (map[string]map[string
 			scanner.Text())
 	}
 
-	header := strings.Fields(parts[1])
+        header_recieve := strings.Fields(parts[1])
+        header_transmit := strings.Fields(parts[2])
+
 	netDev := map[string]map[string]string{}
 	for scanner.Scan() {
 		line := strings.TrimLeft(scanner.Text(), " ")
 		parts := procNetDevFieldSep.Split(line, -1)
-		if len(parts) != 2*len(header)+1 {
+		if len(parts) != len(header_recieve)+len(header_transmit)+1 {
```

Comment:
Let's move `len(header_recieve)+len(header_transmit)+1` up to line 55 as something like this:

```
headerLength := len(recieveHeader)+len(transmitHeader)+1
```

Then we can test `len(parts) != headerLength`.